pkg/api/pod/util.go

```diff
@@ -356,7 +380,7 @@ func dropDisabledFields(
 
 	if !utilfeature.DefaultFeatureGate.Enabled(features.VolumeSubpath) && !subpathInUse(oldPodSpec) {
 		// drop subpath from the pod if the feature is disabled and the old spec did not specify subpaths
-		VisitContainers(podSpec, func(c *api.Container) bool {
+		VisitContainers(podSpec, DefaultContainers, func(c *api.Container, containerType ContainerType) bool {
```

Comment:
drop from all containers, not just feature-enabled ones (applies to all "iterate over all containers to drop a field" calls to VisitContainers)